dashboard/src/actions/charts.ts

```diff
@@ -80,6 +95,26 @@ export function fetchCharts(
   };
 }
 
+export function fetchChartCategories(
+  cluster: string,
+  namespace: string,
+  repos: string,
+): ThunkAction<Promise<IChartCategory[]>, IStoreState, null, ChartsAction> {
```

Comment:
since you are dispatching the categories (and later or retrieving them from the redux state), you don't need this function to return the array: `ThunkAction<Promise<void>`